agent/config/ratelimited_file_watcher.go

```diff
@@ -0,0 +1,85 @@
+package config
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+)
+
+type rateLimitedFileWatcher struct {
+	watcher          Watcher
+	eventCh          chan *FileWatcherEvent
+	coalesceInterval time.Duration
+}
+
+func (r rateLimitedFileWatcher) Start(ctx context.Context) {
```

Comment:
I think all of these receivers should be pointers. In almost every circumstance that's what you want, the non-pointer receiver should be an exception.

```suggestion
func (r *rateLimitedFileWatcher) Start(ctx context.Context) {
```

Reply:
Nice catch, it was a typo